src/test/java/pro/belbix/ethparser/web3/uniswap/UniswapParserTest.java

```diff
@@ -39,6 +42,48 @@
   @Autowired
   private UniToHarvestConverter uniToHarvestConverter;
 
+  @Test
+  public void test_isFullParsableLpAddressAndDate() {
+    boolean one = isFullParsableLpAddressAndDate("0x0", BigInteger.valueOf(10777067),
+        ETH_NETWORK);
+    boolean two = isFullParsableLpAddressAndDate(
+        "0x56feAccb7f750B997B36A68625C7C596F0B41A58", BigInteger.valueOf(11407437), ETH_NETWORK);
+    boolean three =
+        isFullParsableLpAddressAndDate("0x56feAccb7f750B997B36A68625C7C596F0B41A58",
+            BigInteger.valueOf(11407438), ETH_NETWORK);
+    boolean four =
+        isFullParsableLpAddressAndDate("0x56feAccb7f750B997B36A68625C7C596F0B41A58",
+            BigInteger.valueOf(11407436), ETH_NETWORK);
+
+    assertAll(
+        () -> assertFalse(one),
+        () -> assertFalse(two),
+        () -> assertTrue(three),
+        () -> assertFalse(four)
+    );
+  }
+
+  @Test
+  public void test_cheсking_wrong_contract() {
+    List<LogResult> logResultOne = web3Functions
+        .fetchContractLogs(singletonList("0x56feAccb7f750B997B36A68625C7C596F0B41A58"),
+            10777652,
+            10777652, ETH_NETWORK);
+
+    List<LogResult> logResultsTwo = web3Functions
+        .fetchContractLogs(singletonList("0xb9fa44b0911f6d777faab2fa9d8ef103f25ddf49"),
+            11407008,
+            11407008, ETH_NETWORK);
+
+    UniswapDTO dto1 = uniswapLpLogParser.parse((Log) logResultOne.get(0).get(), ETH_NETWORK);
+    UniswapDTO dto2 = uniswapLpLogParser.parse((Log) logResultsTwo.get(0).get(), ETH_NETWORK);
+
+    assertAll(
```

Comment:
use assertAll for checking multiple fields instead of multiple cases
cases should be split to dedicated test methods